src/main/java/reposense/parser/CsvParser.java

```diff
@@ -211,22 +223,51 @@ private String getRowContentAsRawString(final CSVRecord record) {
     }
 
     /**
-     * Checks if {@code possibleHeader} contains the expected number of columns.
-     * @throws InvalidCsvException if {@code possibleHeader} does not have as many columns as expected.
+     * Generates map of column header to position number for input {@code possibleHeader}.
+     * @throws InvalidCsvException if {@code possibleHeader} does not contain all the mandatory headers.
      */
     private void validateHeader(String[] possibleHeader) throws InvalidCsvException {
-        int actualNumberOfColumns = possibleHeader.length;
-        if (actualNumberOfColumns != expectedHeaderSize) {
-            throw new InvalidCsvException(String.format(
-                    MESSAGE_WRONG_HEADER_SIZE, csvFilePath.getFileName(), actualNumberOfColumns,
-                    expectedHeaderSize));
+        headerSize = possibleHeader.length;
+        for (int i = 0; i < headerSize; i++) {
+            String possible = possibleHeader[i].trim();
+            for (String parsedHeader : mandatoryAndOptionalHeaders()) {
+                if (possible.equalsIgnoreCase(parsedHeader)) {
+                    if (headerMap.containsKey(parsedHeader)) {
+                        logger.warning(String.format(
+                                MESSAGE_DUPLICATE_COLUMN_HEADER, parsedHeader, csvFilePath.getFileName()));
+                    }
+                    headerMap.put(parsedHeader, i);
```

Comment:
I think that we should take the first instance instead of the last instance of the same header, as we primarily read from left to right and we would usually register the first instance. Open for discussion.